### PR TITLE
Bugfix/NIJO-1039 fix wrong total count

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchState.cs [hunk 1]
```diff
@@ -85,7 +85,18 @@ static string GetErrors(Dictionary<Suppliers, SupplierAvailabilitySearchState> s
 
             static int GetResultsCount(Dictionary<Suppliers, SupplierAvailabilitySearchState> states)
             {
-                return states.Sum(s => s.Value.ResultCount);
+                var totalCount = states.Sum(s => s.Value.ResultCount);
+                var duplicates = 0;
+
+                if (states.All(s => s.Value.TaskState == AvailabilitySearchTaskState.Completed))
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchState.cs:91]: Imagine that there are 3 results and all of them will have different duplicate report id.
3 - 3 = 0

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchState.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchState.cs:91]: Maybe add tests there?

**Review by mingazhev — APPROVED** [review]

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/AvailabilityStorageTests/WideAvailabilitySearchStateCalculation.cs [hunk 2]
```diff
@@ -35,15 +35,31 @@ public void Provider_search_results_count_should_sum()
...
+                (Suppliers.Illusions, SupplierAvailabilitySearchState.Completed(searchId, new List<string>(),144))
             };
         
             var wideAvailabilitySearchState = WideAvailabilitySearchState.FromProviderStates(searchId, providerStates);
             
             Assert.Equal(169, wideAvailabilitySearchState.ResultCount);
         }
+
+        [Fact]
+        public void Provider_search_results_with_duplicates_count_should_sum()
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/AvailabilityStorageTests/WideAvailabilitySearchStateCalculation.cs:49]: "..._should_sum_unique_results" as a variant

**Review by mingazhev — APPROVED** [review]

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/AvailabilityStorageTests/WideAvailabilitySearchStateCalculation.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/AvailabilityStorageTests/WideAvailabilitySearchStateCalculation.cs]: Fixed

**Review by kirillta — APPROVED** [review]

**Pull request merged**